### DISCUSSION TITLE
Extend duration based single line max length

### DISCUSSION THREAD

**Discussion by ssbeast123** [SubtitleEdit/subtitleedit#8426] [General]
My subtitles have line(only one line, not two) with duration 3s and any single line max length, and I only want to extend the duration of 3s to 4s for any lines with a single line max length > 25.  So is there any way? Thank you!

**Comment by uckthis** [comment]
Options> Setting > Single line length> set to 25
Tools > Break/Split Long lines (it'll break any lines longer than 25 into two)
Go to Edit > Modify Selection > Rule: Exactly Two lines > Apply. 
Then. Tools > Adjust Durations > Adjust via Seconds. 

Then select all and unbreak.

**Comment by ssbeast123** [comment]
Sorry for my lack of content, I have corrected it
Line(only one line, not two)
Thank for reply @uckthis, but i mean Line(only one line, not two), can u help?

**Comment by uckthis** [comment]
From your post, I assumed that your subtitle file only has one lines and no two lines. If that's the case, If you do the first step I stated, some of your lines will break into two, and we'll modify those and then unbreak them and make them into one line. 

If your subtitle has two lines already, you can find an option in Modify Selection that'll fit your needs.

**Comment by ssbeast123** [comment]
![image](https://github.com/SubtitleEdit/subtitleedit/assets/95699874/d75949f3-4121-4d09-a3dc-eeddaf195508)

For example, I have the following subtitle, I only want to work (extend duration) with the first line and all other lines in the subtitle have the same form as it is single line and length > 25, double lines and single lines have length <25 ignore, don't work with them.

**Comment by uckthis** [comment]
In Modify Selection, use Regular Expression and then enter this code: `^(.{25,})$` and after that do the 4th step I mentioned in my original post.

**Comment by ssbeast123** [comment]
Great, it worked. But there is one problem, I see it selects lines < 25 that have {\an8} and < i>(italic), maybe it has identified > 25, including {\an8}, < i>

**Comment by uckthis** [comment]
`^([^{}<\r\n\(\\)]{25,})$`

**Comment by ssbeast123** [comment]
Wow, thank you so much!

**Comment by ssbeast123** [comment]
@uckthis Do you know how to display "gap to next" on the main screen? 
If so, it would be great for manual edit "duration", edit only with "next to gap" >1.5s
![image](https://github.com/SubtitleEdit/subtitleedit/assets/95699874/dc472d19-98a2-4ce5-a244-184ad42a46b9)
![image](https://github.com/SubtitleEdit/subtitleedit/assets/95699874/1c65eed3-ad8b-460a-80da-f32540d3898a)

**Comment by uckthis** [comment]
Sorry, I don't know how.

**Comment by niksedk** [comment]
You can show "gap" column, by right-clicking on the column header, and choose the "Gap" column:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/d10b97f1-58f8-4f48-92b5-326d49b9a9cb)

Also, in Tools - Fix common errors, you can fix short display times:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/62b8c4b7-95f9-452e-ae3d-3957f4471217)
